php/libraries/Candidate.class.inc

```diff
@@ -191,7 +191,7 @@ class Candidate
      *                                      configured to be true.
      * @param string      $sex         sex, either 'Male' or 'Female'
      * @param string|null $PSCID       PSCID specified by the user, if available
-     * @param int|null    $projectID   The project ID, if available
+     * @param int         $projectID   The project ID, if available
```

Comment:
It might be nice to change this signature to actually pass a Project object instead of an ID here, if possible. We'd benefit from built-in validation this way.

Up to you though - I'm not sure how complicated this would be.